deployments/setup-fork.ts

```diff
@@ -0,0 +1,210 @@
+import Contracts from '../components/Contracts';
+import { MAX_UINT256 } from '../utils/Constants';
+import { DeployedContracts, getNamedSigners, isTenderlyFork } from '../utils/Deploy';
+import { NATIVE_TOKEN_ADDRESS } from '../utils/TokenData';
+import { toWei } from '../utils/Types';
+import '@nomiclabs/hardhat-ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import '@tenderly/hardhat-tenderly';
+import '@typechain/hardhat';
+import AdmZip from 'adm-zip';
+import { BigNumber } from 'ethers';
+import { getNamedAccounts, network, tenderly } from 'hardhat';
+import 'hardhat-deploy';
+import { HttpNetworkUserConfig } from 'hardhat/types';
+import path from 'path';
+
+interface EnvOptions {
+    FORK_NAME: string;
+    FORK_RESEARCH: boolean;
+    TENDERLY_PROJECT: string;
+    TENDERLY_USERNAME: string;
+    DEV_ADDRESSES: string;
+}
+
+const {
+    TENDERLY_PROJECT,
+    TENDERLY_USERNAME,
+    DEV_ADDRESSES,
+    FORK_NAME,
+    FORK_RESEARCH: isResearch
+}: EnvOptions = process.env as any as EnvOptions;
+
+const tenderlyNetwork = tenderly.network();
+
+const createTenderlyFork = async () => {
+    console.log('Setting up new fork...');
+
+    await tenderlyNetwork.initializeFork();
+
+    const forkId = tenderlyNetwork.getFork()!;
+
+    setForkId(forkId);
+
+    console.log(`Fork ID: ${forkId}`);
+    console.log();
+
+    return forkId;
+};
+
+const setForkId = (forkId: string) => {
+    tenderlyNetwork.setFork(forkId);
+
+    (network.config as HttpNetworkUserConfig).url = `https://rpc.tenderly.co/fork/${forkId}`;
+};
+
+interface FundingRequest {
+    token: string;
+    amount: BigNumber;
+    whale: SignerWithAddress;
+}
+
+const fundAccount = async (account: string, fundingRequests: FundingRequest[]) => {
+    console.log(`Funding ${account}...`);
+
+    for (const fundingRequest of fundingRequests) {
+        if (fundingRequest.token === NATIVE_TOKEN_ADDRESS) {
+            await fundingRequest.whale.sendTransaction({
+                value: fundingRequest.amount,
+                to: account
+            });
+
+            continue;
```

Comment:
I'd use `else` instead.

Reply:
I generally prefer `return`/`continue` to `else` when possible, since it's easier to follow the code (`return`/`continue` -> no need to think about the previous block anymore)